staging/src/k8s.io/kubectl/pkg/cmd/apiresources/apiresources.go

```diff
@@ -205,18 +229,36 @@ func (o *APIResourceOptions) RunAPIResources() error {
 				APIGroupVersion: gv.String(),
 				APIResource:     resource,
 			})
+			apiResources = append(apiResources, resource)
+		}
+		if len(apiResources) > 0 {
+			apiList.APIResources = apiResources
+			filteredList = append(filteredList, apiList)
 		}
 	}
 
-	if o.NoHeaders == false && o.Output != "name" {
-		if err = printContextHeaders(w, o.Output); err != nil {
+	if o.NoHeaders == false && *o.PrintFlags.OutputFormat != "name" && (*o.PrintFlags.OutputFormat != "json" && *o.PrintFlags.OutputFormat != "yaml") {
```

Comment:
Why don't we do only `*o.PrintFlags.OutputFormat == ""`

Reply:
Modified this, but it needed more checks than that. PTAL.